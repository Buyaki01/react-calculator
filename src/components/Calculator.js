import React from 'react';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

  handleClick = async (e) => {
    const response = await calculate(this.state, e.target.name);
    this.setState(() => ({
      total: response.total,
      next: response.next,
      operation: response.operation,
    }));
  }

  render() {
    const { total, next } = this.state;
    return (
      <div className="allItems">
        <div className="inputBtn">
          {next || total || 0}
        </div>
        <div className="grid-container">
          <button type="button" onClick={this.handleClick} className="grid-item" name="AC"> AC </button>
          <button type="button" onClick={this.handleClick} className="grid-item" name="+/-"> +/- </button>
          <button type="button" onClick={this.handleClick} className="grid-item" name="%"> % </button>
          <button type="button" onClick={this.handleClick} className="grid-item operations" name="รท"> รท </button>
          <button type="button" onClick={this.handleClick} className="grid-item" name="7"> 7 </button>
          <button type="button" onClick={this.handleClick} className="grid-item" name="8"> 8 </button>
          <button type="button" onClick={this.handleClick} className="grid-item" name="9"> 9 </button>
          <button type="button" onClick={this.handleClick} className="grid-item operations" name="x"> x </button>
          <button type="button" onClick={this.handleClick} className="grid-item" name="4"> 4 </button>
          <button type="button" onClick={this.handleClick} className="grid-item" name="5"> 5 </button>
          <button type="button" onClick={this.handleClick} className="grid-item" name="6"> 6 </button>
          <button type="button" onClick={this.handleClick} className="grid-item operations" name="-"> - </button>
          <button type="button" onClick={this.handleClick} className="grid-item" name="1"> 1 </button>
          <button type="button" onClick={this.handleClick} className="grid-item" name="2"> 2 </button>
          <button type="button" onClick={this.handleClick} className="grid-item" name="3"> 3 </button>
          <button type="button" onClick={this.handleClick} className="grid-item operations" name="+"> + </button>
          <button type="button" onClick={this.handleClick} className="grid-item addColSpan" name="0"> 0 </button>
          <button type="button" onClick={this.handleClick} className="grid-item" name="."> . </button>
          <button type="button" onClick={this.handleClick} className="grid-item operations" name="="> = </button>
        </div>
      </div>
    );
  }
}

export default Calculator;
