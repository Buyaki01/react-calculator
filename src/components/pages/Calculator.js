import React, { useState } from 'react';
import calculate from '../../logic/calculate';

function Calculator() {
  const [data, setData] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = async (e) => {
    const response = await calculate(data, e.target.name);
    setData({
      total: response.total,
      next: response.next,
      operation: response.operation,
    });
  };

  const { total, next } = data;
  return (
    <>
      <div className="heading">
        <h1>Math Magicians</h1>
      </div>
      <div className="calculatorSec">
        <div className="firstColumn">
          <h3> Lets do some math! </h3>
        </div>
        <div className="allItems">
          <div className="inputBtn">
            {next || total || 0}
          </div>
          <div className="grid-container">
            <button type="button" onClick={handleClick} className="grid-item" name="AC"> AC </button>
            <button type="button" onClick={handleClick} className="grid-item" name="+/-"> +/- </button>
            <button type="button" onClick={handleClick} className="grid-item" name="%"> % </button>
            <button type="button" onClick={handleClick} className="grid-item operations" name="รท"> รท </button>
            <button type="button" onClick={handleClick} className="grid-item" name="7"> 7 </button>
            <button type="button" onClick={handleClick} className="grid-item" name="8"> 8 </button>
            <button type="button" onClick={handleClick} className="grid-item" name="9"> 9 </button>
            <button type="button" onClick={handleClick} className="grid-item operations" name="x"> x </button>
            <button type="button" onClick={handleClick} className="grid-item" name="4"> 4 </button>
            <button type="button" onClick={handleClick} className="grid-item" name="5"> 5 </button>
            <button type="button" onClick={handleClick} className="grid-item" name="6"> 6 </button>
            <button type="button" onClick={handleClick} className="grid-item operations" name="-"> - </button>
            <button type="button" onClick={handleClick} className="grid-item" name="1"> 1 </button>
            <button type="button" onClick={handleClick} className="grid-item" name="2"> 2 </button>
            <button type="button" onClick={handleClick} className="grid-item" name="3"> 3 </button>
            <button type="button" onClick={handleClick} className="grid-item operations" name="+"> + </button>
            <button type="button" onClick={handleClick} className="grid-item addColSpan" name="0"> 0 </button>
            <button type="button" onClick={handleClick} className="grid-item" name="."> . </button>
            <button type="button" onClick={handleClick} className="grid-item operations" name="="> = </button>
          </div>
        </div>
      </div>
    </>
  );
}

export default Calculator;
