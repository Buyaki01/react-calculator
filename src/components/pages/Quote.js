import React from 'react';

const Quote = () => (
  <div>
    <h3>Mathematics Quotes</h3>
  </div>
);
export default Quote;
