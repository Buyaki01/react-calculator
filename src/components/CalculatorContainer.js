import React from 'react';
import { Route, Routes } from 'react-router-dom';
import Calculator from './pages/Calculator';
import Quote from './pages/Quote';
import Navbar from './Navbar';

const CalculatorContainer = () => (
  <>
    <Navbar />
    <Routes>
      <Route
        path="/"
        element={(
          <div className="container">
            <div>
              <h1>Math Magicians</h1>
            </div>
            <div className="homePage">
              <div className="homePageTitle">
                <h3>Welcome to our page! </h3>
              </div>
              <div className="homePageBody">
                <p className="paragraphs">
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
                </p>
                <p className="paragraphs">
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
                  Lorem Ipsum is simply dummy text of the printing and typesetting industry.
                </p>
              </div>
            </div>
          </div>
        )}
      />

      <Route path="/calculator" element={<Calculator />} />
      <Route path="/quote" element={<Quote />} />
    </Routes>
  </>
);
export default CalculatorContainer;
