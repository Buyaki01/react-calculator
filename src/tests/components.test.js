import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import CalculatorContainer from '../components/CalculatorContainer';
import Calculator from '../components/pages/Calculator';
import Quote from '../components/pages/Quote';

describe('Test UI of the webpage', () => {
  it('Displays the Homepage', () => {
    const { asFragment } = render(
      <CalculatorContainer />,
      { wrapper: MemoryRouter },
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Displays the Calculator page', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Displays the details page', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
