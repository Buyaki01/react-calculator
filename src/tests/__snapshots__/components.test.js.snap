// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UI of the webpage Displays the Calculator page 1`] = `
Array [
  <div
    className="calcHeadingSec"
  >
    <h1>
      Math Magicians
    </h1>
  </div>,
  <div
    className="calculatorSec"
  >
    <div
      className="firstColumn"
    >
      <h3>
         Lets do some math! 
      </h3>
    </div>
    <div
      className="allItems"
    >
      <div
        className="inputBtn"
      >
        0
      </div>
      <div
        className="grid-container"
      >
        <button
          className="grid-item"
          name="AC"
          onClick={[Function]}
          type="button"
        >
           AC 
        </button>
        <button
          className="grid-item"
          name="+/-"
          onClick={[Function]}
          type="button"
        >
           +/- 
        </button>
        <button
          className="grid-item"
          name="%"
          onClick={[Function]}
          type="button"
        >
           % 
        </button>
        <button
          className="grid-item operations"
          name="รท"
          onClick={[Function]}
          type="button"
        >
           รท 
        </button>
        <button
          className="grid-item"
          name="7"
          onClick={[Function]}
          type="button"
        >
           7 
        </button>
        <button
          className="grid-item"
          name="8"
          onClick={[Function]}
          type="button"
        >
           8 
        </button>
        <button
          className="grid-item"
          name="9"
          onClick={[Function]}
          type="button"
        >
           9 
        </button>
        <button
          className="grid-item operations"
          name="x"
          onClick={[Function]}
          type="button"
        >
           x 
        </button>
        <button
          className="grid-item"
          name="4"
          onClick={[Function]}
          type="button"
        >
           4 
        </button>
        <button
          className="grid-item"
          name="5"
          onClick={[Function]}
          type="button"
        >
           5 
        </button>
        <button
          className="grid-item"
          name="6"
          onClick={[Function]}
          type="button"
        >
           6 
        </button>
        <button
          className="grid-item operations"
          name="-"
          onClick={[Function]}
          type="button"
        >
           - 
        </button>
        <button
          className="grid-item"
          name="1"
          onClick={[Function]}
          type="button"
        >
           1 
        </button>
        <button
          className="grid-item"
          name="2"
          onClick={[Function]}
          type="button"
        >
           2 
        </button>
        <button
          className="grid-item"
          name="3"
          onClick={[Function]}
          type="button"
        >
           3 
        </button>
        <button
          className="grid-item operations"
          name="+"
          onClick={[Function]}
          type="button"
        >
           + 
        </button>
        <button
          className="grid-item addColSpan"
          name="0"
          onClick={[Function]}
          type="button"
        >
           0 
        </button>
        <button
          className="grid-item"
          name="."
          onClick={[Function]}
          type="button"
        >
           . 
        </button>
        <button
          className="grid-item operations"
          name="="
          onClick={[Function]}
          type="button"
        >
           = 
        </button>
      </div>
    </div>
  </div>,
]
`;

exports[`Test UI of the webpage Displays the Homepage 1`] = `
<DocumentFragment>
  <nav
    class="navBarSec"
  >
    <ul
      class="navBarLinks"
    >
      <li>
        <a
          aria-current="page"
          class="listItem active"
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          class="listItem"
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          class="listItem"
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
  <div
    class="container"
  >
    <div>
      <h1>
        Math Magicians
      </h1>
    </div>
    <div
      class="homePage"
    >
      <div
        class="homePageTitle"
      >
        <h3>
          Welcome to our page! 
        </h3>
      </div>
      <div
        class="homePageBody"
      >
        <p
          class="paragraphs"
        >
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry.
        </p>
        <p
          class="paragraphs"
        >
          Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum is simply dummy text of the printing and typesetting industry.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test UI of the webpage Displays the details page 1`] = `
<div
  className="quotesSec"
>
  <div
    className="quotesHeading"
  >
    <h1>
      Math Magicians
    </h1>
  </div>
  <div>
    <h3
      className="quotes"
    >
      Mathematics is the art of giving the same name to different things. - Henri Poincare
    </h3>
  </div>
</div>
`;
